cpp/plugins/cucim.kit.cuslide/src/cuslide/jpeg/libjpeg_turbo.cpp

```diff
@@ -196,6 +231,119 @@ bool read_jpeg_header_tables(const void* handle, const void* jpeg_buf, unsigned
     return true;
 }
 
+// tjDecompress2
+int jpeg_decode_buffer(const void* handle,
+                       const unsigned char* jpegBuf,
+                       unsigned long jpegSize,
+                       unsigned char* dstBuf,
```

Comment:
Should we do `nullptr` checks for these below?

Reply:
This method (jpeg_decode_buffer()) is from the original function [tjDecompress2()](https://github.com/libjpeg-turbo/libjpeg-turbo/blob/2.0.6/turbojpeg.c#L1241) and did minimal change to make it work
- line 249-253 are added to replace `GET_DINSTANCE(handle);` macro
- line 282-286 are added to set color space

For other code that is borrowed from the external library, I would like to keep the source code as it is so that further changes from the original code are trackable.

Instead, added some more comments for explaining that.